src/library/scala/collection/immutable/LinkedMap.scala

```diff
@@ -0,0 +1,209 @@
+package scala
+package collection
+package immutable
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+
+import scala.collection.mutable.{Builder, ImmutableBuilder}
+
+/**
+  * A generic trait for ordered immutable maps. Concrete classes have to provide
+  * functionality for the abstract methods in `LinkedMap`:
+  *
+  * @tparam A
+  * @tparam B
+  */
+
+trait LinkedMap[K, +V]
+  extends Map[K, V]
+    with Iterable[(K, V)]
+    with MapOps[K, V, LinkedMap, LinkedMap[K, V]]
+    with Equals
+
+object LinkedMap extends MapFactory[LinkedMap] {
+  def empty[K, V]: LinkedMap[K, V] = EmptyLinkedMap.asInstanceOf[LinkedMap[K, V]]
+
+  def from[K, V](it: collection.IterableOnce[(K, V)]): LinkedMap[K, V] =
+    it match {
+      case vm: LinkedMap[K, V] => vm
+      case _ => (newBuilder[K, V] ++= it).result()
+    }
+
+  def newBuilder[K, V]: Builder[(K, V), LinkedMap[K, V]] =
+    new ImmutableBuilder[(K, V), LinkedMap[K, V]](empty) {
+      def addOne(elem: (K, V)): this.type = { elems = elems + elem; this }
+    }
```

Comment:
I think we should extend `MapFactory.Delegate[LinkedMap](VectorMap)` instead of implementing `empty`, `from` and `newBuilder`.